rskj-core/src/main/java/co/rsk/trie/Trie.java

```diff
@@ -485,7 +486,7 @@ public Trie delete(String key) {
             internalToMessage();
         }
 
-        return cloneArray(encoded);
+        return encoded;
```

Comment:
Is this change really needed for the purpose of this pull request? Yes, cloneArray is a bit of "defensive programming" but really aisle the external provided value from any modification.

Reply:
yes, it creates a lot of arrays' copies during manipulations with a trie. Without this change the import process fails with OoM (with 4gb of ram).

Also we don't defend underlying data / array in other classes, e.g. `Keccak256` and `ByteArrayWrapper`. At the end, this is a tradeoff.